### PR TITLE
Deprecated response_format in vso search

### PR DESCRIPTION
**Pull request by nabobalis**
Supersedes: https://github.com/sunpy/sunpy/pull/7364 

So during the 2.1 days we added this old response type to prevent some breaking changes from the rework of the metadata and we never got around to removing it.
So I have finally marked it as deprecated alongside some keywords that were unused.

This should not be backported.

### CODE REVIEW

File: sunpy/net/hek2vso/hek2vso.py [hunk 1]
```diff
@@ -150,8 +148,10 @@ def full_query(self, client_query, limit=None, progress=False):
         self.hek_results = self.hek_client.search(*client_query)
         self._quick_clean()
         return self.translate_and_query(self.hek_results,
-                                        limit=limit, progress=progress)
+                                        limit=limit)
 
+    @deprecated_renamed_argument("vso_response_format", None,"6.0", warning_type=SunpyDeprecationWarning)
+    @deprecated_renamed_argument("progress", None,"6.0", warning_type=SunpyDeprecationWarning)
```
Comment (nabobalis) [inline sunpy/net/hek2vso/hek2vso.py:154]: The other choice is to pass progress down so it works?

Comment (wtbarnes) [reply sunpy/net/hek2vso/hek2vso.py:154]: Why would we deprecate it in `full_query` but not here? Maybe I'm missing some context

File: changelog/7468.deprecation.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+The keyword ``response_format`` in :meth:`sunpy.net.vso.VSOClient.search` has been deprecated.
+This was originally introduced to restore the previous behaviour of the VSO client, which was to return
+``sunpy.net.vso.legacy_response.QueryResponse`` instead of `sunpy.net.vso.table_response.VSOQueryResponseTable` objects.
+This behaviour has been the default for over 4 years and the keyword is no longer needed.
+The plan is to remove this keyword and the older ``sunpy.net.vso.legacy_response.QueryResponse`` class in a sunpy 7.0.
```
Comment (wtbarnes) [inline changelog/7468.deprecation.rst:5]:
```suggestion
The plan is to remove this keyword and the older ``sunpy.net.vso.legacy_response.QueryResponse`` class in a sunpy 7.0.
The keyword `progress` in :meth:`sunpy.net.hek2vso.H2VClient.full_query` has been deprecated and will be removed in sunpy 7.0.
```

File: sunpy/net/hek2vso/hek2vso.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/net/hek2vso/hek2vso.py:154]: I mean more, I could work out how where progress should be passed and then not deprecate it across everything.

File: changelog/7468.deprecation.rst [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+The keyword `progress` in :meth:`sunpy.net.hek2vso.H2VClient.full_query` has been deprecated and will be removed in sunpy 7.0.
+The keyword `progress` in :meth:`sunpy.net.hek2vso.H2VClient.full_query` has been deprecated and will be removed in sunpy 7.0.
+The keyword `progress` in :meth:`sunpy.net.hek2vso.H2VClient.full_query` has been deprecated and will be removed in sunpy 7.0.
```
Comment (nabobalis) [inline changelog/7468.deprecation.rst:8]:
```suggestion
```
Suggestion UI is broken

**Review by alasdairwilson — APPROVED** [review]

File: changelog/7468.deprecation.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+The keyword ``response_format`` in :meth:`sunpy.net.vso.VSOClient.search` has been deprecated.
+This was introduced to preserve legacy behaviour of the VSO client, to return
+``sunpy.net.vso.legacy_response.QueryResponse`` instead of `sunpy.net.vso.table_response.VSOQueryResponseTable` objects.
+This behaviour has been the default for over 4 years and the keyword is no longer needed.
+This keyword and the older ``sunpy.net.vso.legacy_response.QueryResponse`` class will be removed in sunpy 7.0.
+The keyword `progress` in :meth:`sunpy.net.hek2vso.H2VClient.full_query` has been deprecated and will be removed in sunpy 7.0.
```
Comment (nabobalis) [inline changelog/7468.deprecation.rst:6]:
```suggestion
The keyword ``progress`` in :meth:`sunpy.net.hek2vso.H2VClient.full_query` has been deprecated and will be removed in sunpy 7.0.
```

**Comment by nabobalis** [comment]
Thanks for the reviews @wtbarnes and @alasdairwilson

**Pull request merged**